package/MDAnalysis/analysis/base.py

```diff
@@ -163,3 +167,151 @@ def run(self):
         logger.info("Finishing up")
         self._conclude()
         return self
+
+
+class AnalysisFromFunction(AnalysisBase):
+    """
+    Create an analysis from a function working on AtomGroups
+
+    Attributes
+    ----------
+    results : ndarray
+        results of calculation are stored after call to ``run``
+
+    Example
+    -------
+    >>> def rotation_matrix(mobile, ref):
+    >>>     return mda.analysis.align.rotation_matrix(mobile, ref)[0]
+
+    >>> rot = AnalysisFromFunction(rotation_matrix, trajectory, mobile, ref).run()
+    >>> print(rot.results)
+
+    Raises
+    ------
+    ValueError : if ``function`` has the same kwargs as ``BaseAnalysis``
+    """
+
+    def __init__(self, function, trajectory=None, *args, **kwargs):
+        """Parameters
+        ----------
+        function : callable
+            function to evaluate at each frame
+        trajectory : mda.coordinates.Reader (optional)
+            trajectory to iterate over. If ``None`` the first AtomGroup found in
+            args and kwargs is used as a source for the trajectory.
+        *args : list
+           arguments for ``function``
+        **kwargs : dict
+           arugments for ``function`` and ``AnalysisBase``
+
+        """
+        if (trajectory is not None) and (not isinstance(
+                trajectory, coordinates.base.Reader)):
+            args = args + (trajectory, )
+            trajectory = None
+
+        if trajectory is None:
+            for arg in args:
+                if isinstance(arg, AtomGroup):
+                    trajectory = arg.universe.trajectory
+            # when we still didn't find anythgin
+            if trajectory is None:
+                for arg in six.itervalues(kwargs):
+                    if isinstance(arg, AtomGroup):
+                        trajectory = arg.universe.trajectory
+
+        self.function = function
+        self.args = args
+        base_kwargs, self.kwargs = _filter_baseanalysis_kwargs(self.function,
+                                                               kwargs)
+
+        super(AnalysisFromFunction, self).__init__(trajectory, **base_kwargs)
+
+    def _prepare(self):
+        self.results = []
+
+    def _single_frame(self):
+        self.results.append(self.function(*self.args, **self.kwargs))
+
+    def _conclude(self):
+        self.results = np.asarray(self.results)
+
+
+def analysis_class(function):
+    """decorator to transform a function operating on a single frame to an
```

Comment:
I would formulate it as

```
Transform a function operating on a single frame to an analysis class

For an usage in a library we recommend the following style:

>>> def simple(mobile):
>>>     pass
>>> SimpleAnalysis = analysis_class(simple)

It can also be used as a decorator:

>>> @analysis_class
 >>> def RotationMatrix(mobile, ref):
>>>     return mda.analysis.align.rotation_matrix(mobile, ref)[0]

>>> rot = RotationMatrix(u.trajectory, mobile, ref, step=2).run()
>>> print(rot.results)
```

Then of course, the example would go away.

The point is to have both example together rather than scattered throughout the docstring. Also, the change in the initial description remove the emphasis on the decorator aspect to match with the recommendation just bellow.